app/forms/work_form.rb

```diff
@@ -85,7 +85,11 @@ def sync(*)
                  attached_files.delete(item)
                  return skip!
                end
-               item || attached_files.append(AttachedFile.new)
+               return item if item
+
+               new_attached_file = AttachedFile.new
```

Comment:
This changes makes a connection between AttachedFile instances and AS::Blobs that wasn't happening before.